kpi/serializers/current_user.py

```diff
@@ -81,6 +80,19 @@ def get_git_rev(self, obj):
         else:
             return False
 
+    def get_social_accounts(self, user):
+        private_apps = [
+            app["provider"]
+            for app in SocialApp.objects.filter(
+                custom_data__isnull=False
+            ).values("provider")
+        ]
+        qs = SocialAccount.objects.filter(user=user).exclude(
```

Comment:
I think this would run multiple queries when it could just run one.

Reply:
Yes, you're right. I'll figure out how to simplify this to one query.

Reply:
Actually I don't think we need any filter here either. As I mentioned prior, it's not a problem to know that you yourself are a member of a private SSO app. The only place we need to restrict is the Login/Register page, which comes from /environment/

Reply:
Haha right of course! Will fix this up too.